packages/mcdoc/src/runtime/checker/index.ts

```diff
@@ -1148,9 +1159,25 @@ export function simplify<T>(
 			}
 		case 'enum':
 			return { ...typeDef, enumKind: typeDef.enumKind ?? 'int' }
-		case 'concrete': // TODO
-		case 'template': // TODO
-			return { kind: 'union', members: [] }
+		case 'concrete':
+			return simplify(typeDef.child, options, parents, typeDef.typeArgs)
+		case 'template':
+			if (typeArgs.length !== typeDef.typeParams.length) {
+				options.context.logger.warn(
+					`Expected ${typeDef.typeParams.length} mcdoc type args for ${
+						McdocType.toString(typeDef.child)
+					}, but got ${typeArgs.length}`,
+				)
+			}
+			typeArgs.forEach((arg, i) => {
+				const path = typeDef.typeParams[i]?.path
+				if (!path) return
+				options.context.symbols.query(options.context.doc, 'mcdoc', path)
+					.enter({
+						data: { data: { typeDef: arg } satisfies TypeDefSymbolData },
+					})
```

Comment:
I'm not fully happy with this approach yet. This may cause ideally it should only temporarily affect the symbol table, but I'm not sure how to do that yet. If that turns out impossible, another parameter may need to be passed to `simplify`

Reply:
The symbol table is supposed to be "stable" given the state of the workspace (i.e. a list of files and their content) is not changed so i'm not sure how temporary symbols would behave. Is it possible to handle the substitution of type args in `mcdoc` domain instead of relying on lookups in the symbol table?

Reply:
I see. The issue is that in a template type, types like `::java::data::util::T` are indistinguishable from normal reference types, which made me immediately think that the temporary type arguments should be handled by the symbol table somehow. I realize this assumption is wrong now.